apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common.Tests/Trace/SimpleManagedTracerTest.cs

```diff
@@ -79,6 +84,49 @@ private static bool Is1MsSpan(TraceSpan span)
             return duration.Seconds == 0 && duration.Nanos == 1_000_000;
         }
 
+        /// <summary>
+        /// Used to ensure that <see cref="Timestamp"/>s manipulated by
+        /// <see cref="SimpleManagedTracer"/> are normalized after
+        /// manipulation.
+        /// </summary>
+        private static bool IsNormalized(Timestamp timestamp) =>
+            timestamp.Nanos >= 0 &&
+            timestamp.Nanos <= MaxNanos &&
+            timestamp.Seconds >= UnixSecondsAtBclMinValue &&
+            timestamp.Seconds <= UnixSecondsAtBclMaxValue;
+
+        [Theory]
+        [InlineData(10, 10, 10, 15)]
+        [InlineData(10, 999000000, 10, 999000015)]
+        [InlineData(10, 999999999, 11, 15)]
+        [InlineData(10, 15, 10, 10)]
+        public void Span_EnsureVisibleDuration(int startSeconds, int startNanos, int endSeconds, int endNanos)
+        {
+            var mockConsumer = new Mock<IConsumer<TraceProto>>();
```

Comment:
What's this mock actually doing? If we don't care about how it responds, I think I'd normally create a "no-op" fake instead of mocking.

Reply:
Done